### PR TITLE
New Welcome Section

### PR DESCRIPTION
**Pull request by asleeponmykeyboard**
Filled in all the pages in the welcome section to the best of my ability

If I spelt anything incorrectly or was too wordy with my explanations, let me know

### CODE REVIEW

**Comment by asleeponmykeyboard** [comment]
Should I update every reference to bevy import from 0.5 to 0.6

**Comment by alice-i-cecile** [comment]
Have you seen #176? I think we may have duplicated work 😅

Good to have a second set of eyes; we can review both and compare / merge content.

**Comment by asleeponmykeyboard** [comment]
Oh, sorry, I just saw yours, let's compare work.

File: content/learn/book/welcome/_index.md [hunk 1]
```diff
@@ -6,4 +6,33 @@ template = "book-section.html"
...
+- Intended for use with big teams of people with jobs on the line
+- Stable
+
+## General Recommendations
+
+While this book will provide context for some Rust capabilities, it does NOT include a lot of the best practices that the language and this may inhibit your learning. 
+
+We highly recommend reading some form of Rust documentation. The best resource to choose is probably [The Rust Book](https://docs.rust-lang.org/book), as it provides projects and examples.
+
+However, if you like to get your hands dirty quick, you can try [Rust by Example](https://docs.rust-lang.org/stable/rust-by-example) or [Rustlings](https://github.com/rust-lang/rustlings/).
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/_index.md:36]: I like these additional resources a lot; I think we should aim to add these to #176.

File: content/learn/book/welcome/community/_index.md [hunk 2]
```diff
@@ -5,9 +5,27 @@ template = "book-section.html"
...
-TODO: link to awesome-bevy and discuss crate ecosystem
+More can be found at this website's [community page](https://bevyengine.org/community).
 
-TODO: link to CONTRIBUTING.md and talk about how Bevy is maintained by Cart plus a broad community of volunteers.
-Link to Cart's donation page
+Our community is very active and welcoming.
+
+## Third-Party Crates
+
+Bevy has a lot of third-party crates associated with it, such as [bevy_retrograde](https://github.com/katharostech/bevy_retrograde), [bevy_spicy_networking](https://github.com/CabbitStudios/bevy_spicy_networking) and [heron](https://github.com/jcornaz/heron).
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/community/_index.md:19]: We should be careful not to link to Katharos stuff without *very* clear license disclaimers. They seem like nice folks, but the license is a deal-breaker.

File: content/learn/book/welcome/community/_index.md [hunk 3]
```diff
@@ -5,9 +5,27 @@ template = "book-section.html"
...
+
+These are often published to [crates.io](https://crates.io) and use Bevy's plugin system, meaning technically, at this point, you can make your own thrid-party crates (although, they may not be entirely useful).
+
+There is a whole GitHub repository dedicated to showing awesome bevy assets, aptly named [bevy-assets](https://github.com/bevyengine/bevy-assets). Be sure to check it out!
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/community/_index.md:23]: When talking about bevy-assets, we need to be clear that it includes plugins / crates as well as classical "assets" (sounds, models, sprites). As someone without a background in Unity etc. this was confusing to me at first.

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+page_template = "book-section.html"
++++
+
+So, you've just made a project with Bevy. Great! However, currently, it isn't doing things 'the Bevy way'. Let's fix.
+
+## Bevy App Structure
+
+Bevy programs, which include games, tools and more, are conventially named 'Apps' and currently our Bevy program doesn't really follow the app structure or how bevy apps work.
+
+Below is a basic outline of how your Bevy Apps should look.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:14]: This is an interesting idea to show the structural elements and code organization. Hmm....

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    App::build()
+        .add_system(setup.system())
+        .run();
+}
+```
+
+Bear in mind, the order of everything beyond modules is completely up to you.
+
+In detail:
+- First, you should import your modules. In order for a bevy program to work properly, it is **ESSENTIAL** that you import the bevy prelude, as this contains the App struct and other things required for bevy to work.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:61]: This is a bit confusing: you can also manually import each needed element, and will definitely trip up advanced programmers.

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+- Secondly, you should define your components and implementations. As displayed, a component can **optionally have values and implementations** which can be used when creating an entity
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:63]: When writing docs like this, it's important to try and explain what terms mean when you get a good chance, rather than just repeating jargon.

In this case, "implementations" are something of a specialized term for Rust: mentioning that they're "methods" (if you're targeting other programmers) or "functions that belong to a specific type" (if you're targeting complete novices) goes a long way to unobtrusively clarify a concept.

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+- Finally, we have the main function which defines a Bevy App, adds any systems, builds and runs our new App. Whenever, you define a new system, you need to add it to this decleration. This can get messy quick, which is why you usually seperate these things into **plugins** (talked about in a later chapter).
+
+A lot of this, you don't need to wory about right now. Just remember that:
+- You add systems to your app in the main function and
+- You define systems that can act on components
+
+## Making A Bevy App
+
+That was a lot of information to take in, but don't worry. Let's start off simple with a basic Bevy App!
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:75]: That information overload is a sign that the structure is inverted IMO :)

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+- Secondly, you should define your components and implementations. As displayed, a component can **optionally have values and implementations** which can be used when creating an entity
+
+- Thirdly, you should define your systems which are the logic of the game. In this example, we have a `setup()` function which spawns a new entity and inserts 2 componenets, Player and Health. We can use implementations in functions in the insertion. See how it links together!
+
+- Finally, we have the main function which defines a Bevy App, adds any systems, builds and runs our new App. Whenever, you define a new system, you need to add it to this decleration. This can get messy quick, which is why you usually seperate these things into **plugins** (talked about in a later chapter).
+
+A lot of this, you don't need to wory about right now. Just remember that:
+- You add systems to your app in the main function and
+- You define systems that can act on components
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:71]:
```suggestion
- You define systems that can act on the component data of entities
```

The interplay with entities is important here: you're not just operating on columns of data at once, and talking about how entities have components ties the mental model together a bit better.

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+Firstly open `main.rs` in the project from last page and import the Bevy prelude:
+
+```rust
+use bevy::prelude::*;
+
+// Rest of code
+```
+
+Then, replace the main function with a Bevy App Decleration:
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:85]:
```suggestion
Then, initialize and run your Bevy app:
```

File: content/learn/book/welcome/hello-bevy/_index.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+```rust
+fn main() {
+    App::build()
+        .add_system(hello_bevy.system())
+        .run()
+}
+```
+
+We just ran a **trait extension method** which allows us to convert our simple function to a powerful system.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/hello-bevy/_index.md:130]: Oh *that's* a wonderful term of art that we should throw into the ECS chapter.

File: content/learn/book/welcome/plugins/_index.md [hunk 11]
```diff
@@ -5,12 +5,76 @@ template = "book-section.html"
...
-TODO: Explain how plugins work
+Writing all your Bevy code in one file would be a pain. Even a game as simple as snake could take up more than 300 lines of code. That's what plugins are for.
+
+Plugins are essentially collections of systems, components and resources that can be added to the main app. For example, if you want a window to display, you can add the `WindowPlugin` and the `RenderPlugin`.
+
+You can also define your own plugins to seperate related pieces of information into dedicated areas. Finally, there are also plugin groups that group plugins together. 
 
 ## `MinimalPlugins`
 
+One of the aformentioned plugin groups is the `MinimalPlugins` group. This plugin is for the most bare-bones of games or tools. It only includes bevy-core and bevy-app. If you are making a text-based game, this is the option for you.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/plugins/_index.md:16]: Adding links to the documentation (on docs.rs usually) helps curious readers dig into the details of the topic without distracting from the main flow.

File: content/learn/book/welcome/plugins/_index.md [hunk 12]
```diff
@@ -5,12 +5,76 @@ template = "book-section.html"
...
+
+You can add it like so:
+
+```rust
+App::build()
+    .add_plugins(MinimalPlugins)
+    .run()
+```
+
+**NOTE: The Bevy Prelude must be included in `main.rs`**
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/plugins/_index.md:26]: We should ensure that examples are compileable / runnable when possible: this helps beginners and allows us to automatically check for staleness.

The bit of repeated boilerplate is worth it.

File: content/learn/book/welcome/why-bevy/_index.md [hunk 13]
```diff
@@ -5,8 +5,71 @@ template = "book-section.html"
...
+There are plently of great reasons to use Bevy, but here are just a few:
+
+### ECS 
+**TL;DR: Better workflow and abstraction in your code.**
+
+**LR:**
+
+Bevy has been designed to use a flexible data system in order to provide a smart, usable workflow. The engine uses an **Entity Component System** (ECS) to acheive this goal. In traditional game engines, a singular object (entity) contains its logic and components in one system. This is known as an object-oriented entity system.
+
+This has its benifits, however, in general, it is being phased out by ECSs. Instead of containing all the systems and components in a singular entity, in ECS, all of this is abstracted into structs (in Rust) and functions. Entities are defined as unique integers.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/why-bevy/_index.md:17]: > it is being phased out by ECSs

I *personally* think this is likely to be true, but is probably a bit inflammatory :p

> Entities are defined as unique integers

This is mostly an implementation detail: you could use strings or chars or PNG files as entities just fine if you *really* wanted. The critical bit is that they're unique identifiers that you can use to organize and quickly look up the data attached to a particular "unit", whether that's a tangible game object or something more abstract.

File: content/learn/book/welcome/why-bevy/_index.md [hunk 14]
```diff
@@ -5,8 +5,71 @@ template = "book-section.html"
...
+
+**LR:**
+
+Bevy has been designed to use a flexible data system in order to provide a smart, usable workflow. The engine uses an **Entity Component System** (ECS) to acheive this goal. In traditional game engines, a singular object (entity) contains its logic and components in one system. This is known as an object-oriented entity system.
+
+This has its benifits, however, in general, it is being phased out by ECSs. Instead of containing all the systems and components in a singular entity, in ECS, all of this is abstracted into structs (in Rust) and functions. Entities are defined as unique integers. 
+
+Components are structs that contain values for said components. Systems are Rust functions that are parsed into the main function and can specify components that an entity must have to be affected by the system.
+
+**This can allow for a great amount of abstraction that help with S.O.L.I.D principles.**
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/why-bevy/_index.md:21]: This sentence doesn't add a ton of concrete value, and many readers won't know what S.O.L.I.D. is.

The overall idea of summarizing the section is good though.

File: content/learn/book/welcome/why-bevy/_index.md [hunk 15]
```diff
@@ -5,8 +5,71 @@ template = "book-section.html"
...
+
+This can potentially reduce the size of your application or game and give the engine even more functionality than built-in.
+
+### Hot Reloading
+
+**TL;DR: Easy file changes without recompilation**
+
+**LR:** 
+
+The Bevy Game Engine uses a **scene system** that allows for **hot reloading**. In very short, not technological terms, you have the ability to instantly change certain files (meshes, textures, scenes) and view the changes without recompiling.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/why-bevy/_index.md:41]: Non-technical readers won't know what recompiling is ;)

File: content/learn/book/welcome/why-bevy/_index.md [hunk 16]
```diff
@@ -5,8 +5,71 @@ template = "book-section.html"
...
+
+This allows for quick iteration for artists and 3D modelers as well as less compile times.
+
+### Much more!
+
+There is much more Bevy is unique for, including,
+- Fast Compile Times
+- Modular, parallel render graph
+- Flex Box Inspired UI system
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/why-bevy/_index.md:50]: I'm not sure this feature is ready enough to be advertised (nor is flex-box UI very unique, even among game engines).

File: content/learn/book/welcome/why-bevy/_index.md [hunk 17]
```diff
@@ -5,8 +5,71 @@ template = "book-section.html"
...
+
+### MASSIVE stability problems
+
+**TL;DR: API changes for days!**
+
+**LR:**
+
+Bevy is a relatively new engine, going public as early as August 2020, meaning that there is a **VERY** high chance vital parts of the engine can change almost every month (if not every week), especially if you are using the master branch of the source. This is why we recommend holding onto a specific hash when working on a project (we'll get into that later).
+
+Using the engine for major projects as of right now is **HIGHLY DISCOURAGED** (at least until version 1.0 comes out). Trust us when we say the engine changes every update.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/why-bevy/_index.md:65]: I expect we'll hit practical, useful stability well before 1.0 :)

File: content/learn/book/welcome/why-bevy/_index.md [hunk 18]
```diff
@@ -5,8 +5,71 @@ template = "book-section.html"
...
+
+### Not feature-complete
+
+**TL;DR: Half the features ain't in yet.**
+
+**LR:**
+
+This links back to the first point. The engine is so early in development that there is no editor and no mobile support, along with other things. So if that is a necessity in your project, we are currently a no-go.
```
Comment (alice-i-cecile) [inline content/learn/book/welcome/why-bevy/_index.md:73]: We have some mobile support! Audio, UI and animations are the biggest practical blockers right now IMO.

**Comment by alice-i-cecile** [comment]
Things I definitely think we should incorporate:
- link to Rust in Action and Rustlings
- explain how the system magic uses **trait extension methods** (but over in the ECS chapter, #182)

Are you okay to work off my branch @asleeponmykeyboard? It's a fair bit further along. Also come say hi on [the Discord server](https://discord.com/invite/bevy) if you're able (I'm Alice on there) and we can coordinate a bit more casually.

**Comment by asleeponmykeyboard** [comment]
@alice-i-cecile of course, i'll leave this open for now, but tell me if i need to close it

**Comment by asleeponmykeyboard** [comment]
Closing because it's duplicate work. Check out #176 if you wanna contribute to welcome chapter.
